apps/www/_blog/2025-05-08-event-triggers-wo-superuser.mdx

```diff
@@ -0,0 +1,119 @@
+---
+title: PostgreSQL Event Triggers without superuser access
+description: Using Postgres Hooks to allow regular users to create event triggers
+author: steve_chavez
+author_url: https://github.com/steve-chavez
+author_image_url: https://github.com/steve-chavez.png
+image: 2025-05-08-event-triggers-wo-superuser/cover-event-triggers.png
+thumb: 2025-05-08-event-triggers-wo-superuser/cover-event-triggers.png
+categories:
+  - postgres
+tags:
+  - postgres
+  - planetpg
+  - event triggers
+  - extensions
+date: '2025-05-08'
+---
+
+Event triggers in PostgreSQL are a powerful feature but to create them you need to be a SUPERUSER. And giving SUPERUSER to end users isn’t an option in cloud deployments.
+
+Fortunately PostgreSQL is extensible, and thanks to hooks, we can allow end users to create event triggers safely.
```

Comment:
Fortunately, PostgreSQL’s extensibility and hook system make it possible to let end users safely create event triggers.